### PR TITLE
Fix incorrect optional indicators with several parameters

### PR DESCRIPTION
**Pull request by SvenAelterman**
## Description

Several parameters were surrounded with `[ ]`, but they were not optional.

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Sql/Sql/help/Get-AzSqlDatabaseReplicationLink.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Gets the geo-replication links between an Azure SQL Database and a resource grou
 
 ```
 Get-AzSqlDatabaseReplicationLink [-DatabaseName] <String> -PartnerResourceGroupName <String>
- [-PartnerServerName <String>] [-ServerName] <String> [-ResourceGroupName] <String>
+ -PartnerServerName <String> -ServerName <String> -ResourceGroupName <String>
```
[the remark this answers is not in the document]
Comment (isra-fel) [reply src/Sql/Sql/help/Get-AzSqlDatabaseReplicationLink.md]: Hey @SvenAelterman thanks for you contribution! Could you tell us more about what made you think these parameters were optional?

[the remark this answers is not in the document]
Comment (isra-fel) [reply src/Sql/Sql/help/Get-AzSqlDatabaseReplicationLink.md]: > @isra-fel I didn't think they were optional, on the contrary. They were marked as optional, but they aren't.

Sorry I mean why did you think they are mandatory? Did you get errors when not providing these parameters?

[the remark this answers is not in the document]
Comment (isra-fel) [reply src/Sql/Sql/help/Get-AzSqlDatabaseReplicationLink.md]: @maboja-msft @ziwa-msft could you give us some insights about this? Are these parameters intentionally optional, or should they be mandatory? Thanks

[the remark this answers is not in the document]
Comment (SvenAelterman) [reply src/Sql/Sql/help/Get-AzSqlDatabaseReplicationLink.md]: > Sorry I mean why did you think they are mandatory? Did you get errors when not providing these parameters?

Here's why: 
PS screenshot

[the remark this answers is not in the document]
Comment (SvenAelterman) [reply src/Sql/Sql/help/Get-AzSqlDatabaseReplicationLink.md]: PS: I did not remove the optional indicators from the `DatabaseName` parameter name, but I should have done that too.

**Comment by isra-fel** [comment]
@SvenAelterman After some research... here's the thing:

First, the syntax is correct:
```powershell
Get-AzSqlDatabaseReplicationLink [-ResourceGroupName] <string> [-ServerName] <string> [-DatabaseName] <string>
    -PartnerResourceGroupName <string> [-PartnerServerName <string>] [-DefaultProfile <IAzureContextContainer>]
    [-WhatIf] [-Confirm]  [<CommonParameters>]
```
For example: `[-ResourceGroupName] <String>` means that the parameter name `-ServerName` is not required, but the parameter itself is mandatory. [More reference here](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parameters?view=powershell-6#parameter-position)

As a result, when you call a cmdlet by omitting the parameter names, be cautious about the order of the arguments.

Secondly, the reason why you get this error is because: although `PartnerServerName` was designed to be optional, we have a bug in our logic that causes the error when it's not specified. I have filed an [issue](https://github.com/Azure/azure-powershell/issues/10485) on this.

Thanks for you contribution again. I'll close this PR, please track the issue instead :)

**Pull request closed without merging**